src/lib/service.js

```diff
@@ -0,0 +1,26 @@
+
+  function getWeather (city){
```

Comment:
First, let us separate fetching the data from the API from the processing of such data. Why? it's because we want to make sure we can easily change between APIs and easily debug should things go awry with the fetching.

I suggest a better and more proper way to write this asynchronous function:

```javascript
const fetchWeather = async (city) => {
   const API_KEY = 'a622921a4c100b7d35d888ae68406706';
   const url =  `http://api.openweathermap.org/data/2.5/weather?q=${city}& APPID=${API_KEY}`;

  const data = await fetch(url, {mode: 'cors'})
                        .then(response => {
                           if(response.status === 200) return response.json();
                           else return Promise.reject("Cannot find data for city");
                        });
  return data;
}
```
Then you can have another function that processes such data coming from the above:

```javascript

const getWeather = async (city) => {
    const data = await fetchWeather(city);

    const { main, name } = data;
    const { temp } = main;
    
    const Celcius = (temp - 273.15).toFixed(1);
    const Fahrenheit = ((temp - 273.15) * (9 / 5) + 32).toFixed(1);
    const weather = { c:Celcius, f:Fahrenheit, city: name };

    return weather;
  }
```

Reply:
thanks for the review, i see your point.